File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Map<Character,Integer> mp=new HashMap<>();
        int x=0;
        int y=0;
        for(int i=0;i<s.length();i++){
            char c=s.charAt(i);
            if(mp.containsKey(c)){
                x=  Math.max(x,mp.get(c)+1);
            }
            mp.put(c,i);
            y=Math.max(y,i-x+1);
        }
        return y;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        map = {}
        x = 0
        y = 0

        for i in range(len(s)):
            c = s[i]

            if c in map and map[c] >= x:
                x = map[c] + 1

            map[c] = i
            y = max(y, i - x + 1)

        return y